components/external-services-mock/internal/ord-aggregator/handler.go

```diff
@@ -1,21 +1,91 @@
 package ord_aggregator
 
 import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
 	"net/http"
+	"sync"
 
 	"github.com/kyma-incubator/compass/components/external-services-mock/internal/httphelpers"
 	"github.com/pkg/errors"
 )
 
-func HandleFuncOrdConfig(rw http.ResponseWriter, req *http.Request) {
+type ordHandler struct {
+	mutex    sync.Mutex
+	secured  bool
+	username string
+	password string
+}
+
+func NewORDHandler() *ordHandler {
+	return &ordHandler{
+		mutex:   sync.Mutex{},
```

Comment:
You can make it Read Write Lock and lock `HandleFuncOrdConfig` for Read and `HandleFuncOrdConfigSecurity` for Write in order to mitigate many readers lock each other.